src/openeo_aggregator/backend.py

```diff
@@ -691,7 +742,7 @@ def merge(formats: dict, to_add: dict):
                 _log.warning(f"Failed to get service_types from {con.id}: {e!r}", exc_info=True)
                 continue
             # TODO #1 smarter merging:  parameter differences?
-            merge(service_types, types_to_add)
+            merge(service_types, con.id, types_to_add)
```

Comment:
I would even simplify this (for now) by putting the for loop here directly instead of working through this `merge` function, which makes it a bit harder to decipher what is going on